libexec/mcd/mcd-duty

```diff
@@ -0,0 +1,33 @@
+#!/usr/bin/env bash
+### mcd-duty -- Calculate a system duty
+###
+### https://docs.makerdao.com/smart-contract-modules/rates-module
+###
+### Usage: mcd duty [<percent>]
+###
+### example:
+###
+### $ mcd duty 4
+### > 1000000001243680656318820312
+set -e
+
+require-percent() {
+  [ -n "$1" ] || mcd --fail "mcd-duty: Please specify a percentage (i.e. 4.25 == 4.25%)"
+  amount=$(printf "%.0f" "$(echo "scale=0; $1*100" | bc)")
+  [ "$amount" -le "10000" ] && [ "$amount" -ge "0" ] || mcd --fail "mcd-duty: Percentage must be between 0% and 100%"
+}
+
+require-numeric() {
+    re='^[0-9]+([.][0-9]+)?$'
+    if ! [[ $1 =~ $re ]] ; then
+        mcd --fail "mcd-duty: Please enter a numeric value"
+    fi
+}
+
+require-numeric "$1"
+require-percent "$1"
+
+namount="$( echo "scale=4; $amount/10000 + 1" | bc)"
```

Comment:
Shouldn't `10000` actually be `100` here? E.g.
```
$ echo "scale=4; 4/10000 + 1" | bc
1.0004
```
but for 4% I'd expect 1.04.

Reply:
It's not using the user input here. The `require-amount` function scales and sets `$amount` into basis points so we can do some of the comparisons against integers. So `$amount` should be `400` by the time it hits this line.

Reply:
Ah I see it now...thought those were pure functions. Not really a fan of defining a new variable within a function like that, especially when the function has a name that sounds like it's just doing a check. I'd consider returning instead, or maybe just doing away with the functions and using comments to document instead of function names.

Reply:
Yeah, tbh this is was just following some other patterns in here. I'll clean this up to be a simple script.

Reply:
cleaned.

Reply:
Much nicer :).